src/system/SystemLayerImplFreeRTOS.cpp

```diff
@@ -215,5 +233,175 @@ CHIP_ERROR LayerImplFreeRTOS::HandlePlatformTimer()
     return CHIP_NO_ERROR;
 }
 
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS_PLATFORM
```

Comment:
These functions seem to be copy pased from the socket implementation - we should try to leverage them with inheritance instead of copying the full implementation.

Reply:
the implementations differ from select, since in freertos just registering the callback is enough

Reply:
At glance, not all function differ - only seem do. This goes back to my inheritance comment to avoid copy pasting everything.

Reply:
the inheritance will cause a diamond problem since we need API from both FreeRTOS and select. This can be planned and done correctly on the CSA directly.